### PR TITLE
Mar-18-ads-filters

### PR DESCRIPTION
**Pull request by OleksandrProkopenkoDev**
demo https://youtu.be/Yz4XkL13oMY
doc https://oghakirichenko09.atlassian.net/wiki/spaces/~5b055768cd199d3a07c7034e/pages/21168342/Find+Ads+with+Filters

### CODE REVIEW

File: src/main/java/ua/tc/marketplace/model/dto/ad/AdDto.java [hunk 1]
```diff
@@ -20,29 +20,18 @@
...
+        @Size(max = 100, message = "Title cannot be longer than 100 characters")
+        String title,
+    @NotEmpty(message = "Description cannot be empty") String description,
     @NotNull(message = "Price cannot be null")
-    @DecimalMin(value = "0.0", inclusive = false, message = "Price must be greater than zero")
-    BigDecimal price,
-
+        @DecimalMin(value = "0.0", message = "Price must be zero or positive")
+        BigDecimal price,
     List<Photo> photos,
```
Comment (vosmerkin) [inline src/main/java/ua/tc/marketplace/model/dto/ad/AdDto.java:38]: i suggest to use "List<Long> photoIds" here

Comment (OleksandrProkopenkoDev) [reply src/main/java/ua/tc/marketplace/model/dto/ad/AdDto.java:38]: explain why do you think so? `Photo` is just json object, it does not contain files. And do you think it is still redundant?

Comment (vosmerkin) [reply src/main/java/ua/tc/marketplace/model/dto/ad/AdDto.java:38]: Maybe I'm digging too deep. Even without files, I guess there will be many different situations for sending dto, and not every situation needs to have full info. Especially pathes of the files. That is absolutely redundant info for frontend and PhotoDto should be used. But at first place it should be discussed with front end devs, lets leave it for later. Question is off for now

File: docs/endpoints/ad/find-ads-with-filters.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- attribute_breed:Retriever
+- attribute_age:2
+- attribute_size:Big
+- attribute_gender:Male
+- attribute_coat_length:Medium
+- attribute_color:Grey
+- attribute_health_condition:Healthy
+- attribute_pet_name:Rocky
+- page:0
+- size:5
```
Comment (OleksandrProkopenkoDev) [inline docs/endpoints/ad/find-ads-with-filters.md:32]: test comment

**Review by vosmerkin — APPROVED** [review]

**Pull request merged**